### ISSUE TITLE
GAPS: Link stock data to specific storage temperature ranges

### ISSUE THREAD

**Issue by adamdewey** [msupply-foundation/open-msupply#4367]
## Is your feature request related to a problem? Please describe 👀
 
Part of Epic https://github.com/msupply-foundation/open-msupply/issues/4362

## Describe the solution you'd like 🎁

You can already assign multiple locations to a cold chain asset:

![Image](https://github.com/user-attachments/assets/03881146-c99a-4f75-ae39-167d8b01c8c8)

So just need to add temperature_range to a location. This would need to be stored in OMS central, on the `location` table (much like the `name_oms_fields` extension of the `name` table)

Show the temperature range in brackets after the location name, when selecting the location for a stock line (i.e. in inbound shipment)

This can then be used in dashboard queries

### Additional context 💌

| User story  | A.6.2 |
| :------------- | :------------- |
As a… | Health Care Worker (HCW) or warehouse worker
I want… | Stock inventory data to be associated with the correct cold storage location
So that… | I can see exactly what stock each CCE is storing at any given time 

**Solution A.6.2**

The system will allow for users to make the association between stock storage locations and CCE by providing a mechanism to create new stock storage locations that represent the cold storage within CCE.

Multiple locations will be possible for each CCE and the user will be able to specify which type of cold storage they are associated with - e.g. +5 °C and -20 °C for hybrid refrigerator / freezers.

Note: the `exactly what stock each CCE is storing` would be nice but not required at this stage 🙏 

### Moneyworks Jobcode 🧰

`CCEIDEV`

**Comment by lache-melvin** [comment]
Some decision changes since above!

- GAPS script will configure 3 `location_type` records in OG, for +5, -20 and -70
- Create new translation for `location_type` in OMS (OG central data pull)
- Add `location_type_id` to the `location` sync translations
- Add location_type dropdown to the location edit modal in OMS
- Display the location_type description (/name?) in parentheses in the location select dropdowns

**Comment by mark-prins** [comment]
Thanks Laché that's great! 
There is some practical cross-over with the storage capacity properties of cold chain equipment:

<img width="223" alt="image" src="https://github.com/user-attachments/assets/784bacad-26b0-48b3-92d4-dc3e607a0c40">

and

![image](https://github.com/user-attachments/assets/a4f17c88-25d5-43f4-bf13-9d6113fb6798)

We've talked about implementing a dashboard or report to show available volume and volume used. The question is whether we bring across the location volume from OG, or use the volume property from the asset. While not ideal, we can make do with the latter for now.

**Comment by jmbrunskill** [comment]
> We've talked about implementing a dashboard or report to show available volume and volume used. The question is whether we bring across the location volume from OG, or use the volume property from the asset.

I think there's potential for locations to be broken down more than what's in the asset.
E.g. you could a location for each shelf in a fridge. Ideally the system should warn or stop the user from a assigning more volume that the asset actually has. I agree we can live with using asset volumes for now, but thought I'd mention...

**Comment by adamdewey** [comment]
>  Ideally the system should warn or stop the user from a assigning more volume that the asset actually has.

Agree, also ideally the system would actually create a stock location with the appropriate temperature range when creating an asset..

**Comment by CarlosNZ** [comment]
#5179

**Comment by CarlosNZ** [comment]
Just re-opening until I add in the front-end mutation to update the cold storage setting on location

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.4.3`

- [ ] Temperature range shows on the "Locations" chips in Equipment Detail page
- [ ] Temperature range shows on "Locations" dropdown in Equipment Details page
- [ ] Cold storage type info shows on "Locations" inventory table

**Comment by marthakeezy** [comment]
# 🧪 Testing



- [ ] Go to Inventory -> Locations
- [ ] Click a location that has no "Storage type"
- [ ] Can add a storage type in the Edit Location modal that pops up
- [ ] Can change a storage type on a location that already has one
- [ ] Can add a storage type when creating a new location ("New Location" button at the top)
- [ ] Can remove storage type by clearing the input field and saving
__________

# 🧪 Testing



- [ ] Temperature range shows on the "Locations" chips in Equipment Detail page
- [ ] Temperature range shows on "Locations" dropdown in Equipment Details page
- [ ] Cold storage type info shows on "Locations" inventory table

**Comment by marthakeezy** [comment]
hey team! @lache-melvin @adamdewey @CarlosNZ 

apologies for the late testing of this one - i am just going through all backlogged issues from previous releases. 

testing steps above say temperature range should show in equipment detail, however i cannot see it

<img width="506" alt="Image" src="https://github.com/user-attachments/assets/d96d1789-31ed-4cf3-9f7f-4ae695055a5e" />

just confirming this is intended behaviour?

**Comment by adamdewey** [comment]
Thanks @marthakeezy - yes ideally the temperatures would show here

**Comment by marthakeezy** [comment]
reopening as temperature ranges are not showing when selecting cold storage types in locations

**Comment by jmbrunskill** [comment]
> reopening as temperature ranges are not showing when selecting cold storage types in locations

This was a PR request to remove this...
`Per product chat, change the location picker to show the name instead of min - max`


I'm not sure if it was miss-interpreted though.

**Comment by adamdewey** [comment]
Ah, perhaps couldn't have both due to limited space?

If one or the other then probably more useful to stick with the name

**Comment by alainsussol** [comment]
@jmbrunskill 

Hi James, 

Just to make sure. After being re-opened, this issue was closed without a fix and doesn't need to be build tested again ?

**Comment by jmbrunskill** [comment]
@alainsussol sorry I closed it without a comment.
I closed this based on Adam's comment above that with limited space we should stick with the name.
It might be possible to have both on desktop but with mobile view being a priority for GAPS I think just showing the name is fine.
No need to do any testing.

**Comment by marthakeezy** [comment]
Oki - moving to done since the rest of testing has been completed in February :)
